usaspending_api/search/v2/views/spending_over_time.py

```diff
@@ -116,21 +119,26 @@ def apply_elasticsearch_aggregations(self, search: TransactionSearch) -> None:
         Takes in an instance of the elasticsearch-dsl.Search object and applies the necessary
         aggregations in a specific order to get expected results.
         """
-        interval = "year" if self.group == "fiscal_year" else self.group
+        interval = "year" if (self.group == "fiscal_year" or self.group == "calendar_year") else self.group
 
         # The individual aggregations that are needed; with two different sum aggregations to handle issues with
         # summing together floats.
-        group_by_time_period_agg = A(
-            "date_histogram", field="fiscal_action_date", interval=interval, format="yyyy-MM-dd"
-        )
+        group_by_time_period_agg = None
+        if self.group == "calendar_year":
```

Comment:
Is the only thing different in these two condition options the `field`? If so, could you calculate the `field` in the condition and build the agg outside of it, to reduce code duplication?

Reply:
Yes done.